src/users/users.service.ts

```diff
@@ -14,17 +16,48 @@ export class UsersService {
     private usersRepository: Repository<User>,
   ) {}
 
-  async create(createUserInput: CreateUserInput): Promise<User> {
+  async create(createUserInput: CreateUserInput, user: User): Promise<User> {
     try {
+      if (user.role !== Roles.ADMIN) {
+        throw new HttpException(
+          "You don't have permissions",
+          HttpStatus.UNAUTHORIZED,
+        );
+      }
       if (!createUserInput.login) {
         createUserInput.login = createUserInput.firstName + uuid();
       }
       const hash = await bcrypt.hash(createUserInput.password, 12);
-      const user = await this.usersRepository.save({
+      const createdUser = await this.usersRepository.save({
         ...createUserInput,
         password: hash,
       });
-      return user;
+      return createdUser;
+    } catch (err) {
+      throw new HttpException(err, HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  async edit(userId: string, editUserInput: EditUserInput, user: User) {
+    try {
+      if (user.role !== Roles.ADMIN) {
+        throw new HttpException(
+          "You don't have permissions",
+          HttpStatus.UNAUTHORIZED,
+        );
+      }
+      console.log(typeof editUserInput.password);
+
+      if (typeof editUserInput.password === 'string') {
```

Comment:
Do you expecting some other type than string or this is a check for null? 
Also method can be simplified to have just one update and return operation

if(editUserInput.password) {
    editUserInput.password = await bcrypt.hash(editUserInput.password, 12);
}

await this.usersRepository.update(userId, editUserInput);
return await this.usersRepository.findOne(userId);

Reply:
Maybe there, I had just forgotten to delete this, because I was debugging this code line, but the problem wasn't there